echemdb/data/cv/entry.py

```diff
@@ -175,197 +190,148 @@ def names(_, context, role):
+
+                title = tag("i")[field(which_field)]
                 return sentence[title] if as_sentence else title
 
-        return EchemdbStyle(abbreviate_names=True).format_entry("unused", self.bibliography).text.render_as(backend)
+        return (
+            EchemdbStyle(abbreviate_names=True)
+            .format_entry("unused", self.bibliography)
+            .text.render_as(backend)
+        )
 
-    def x(self):
-        r"""
-        Return the name of the variable on the x-axis, i.e., ``"E"``.
+    def field_unit(self, field_name):
+        """Returns the unit of a given field name of the `echemdb` resource.
 
         EXAMPLES::
 
             >>> entry = Entry.create_examples()[0]
-            >>> entry.x()
-            'E'
+            >>> entry.field_unit('E')
+            'V'
 
         """
-        from astropy import units as u
-
-        axis = getattr(self.data_description, 'axes', {})['E']
+        return self.package.get_resource('echemdb').schema.get_field(field_name)['unit']
 
-        if not u.Unit(axis.unit).is_equivalent('V'):
-            raise ValueError("The variable on the x-axis denoted as 'E', is not convertible to 'V'.")
-        
-        return 'E'
-
-    def y(self):
-        r"""
-        Return the name of the variable on the y-axis, i.e., ``"j"`` or ``"I"``.
-
-        EXAMPLES::
+    def rescale_original(self):
+        r"""Returns an entry with a rescaled `echemdb` resource 
+        with the original axes units of the digitized plot. 
 
             >>> entry = Entry.create_examples()[0]
-            >>> entry.y()
-            'j'
+            >>> rescaled_entry = entry.rescale_original()
+            >>> rescaled_entry.package.get_resource('echemdb').schema.fields # doctest: +NORMALIZE_WHITESPACE
+            [{'name': 't', 'unit': 's', 'type': 'number'},
+            {'name': 'E', 'unit': 'V', 'reference': 'RHE', 'type': 'number'},
+            {'name': 'j', 'unit': 'mA / cm2', 'type': 'number'}]
 
         """
-        from astropy import units as u
-
-        axes = getattr(self.data_description, 'axes', {})
-
-        variables = [variable for variable in dir(axes) if variable in ['I', 'j']]
-
-        if len(variables) != 1:
-            raise KeyError("Exactly one of the axes must have variable 'I' or 'j'.")
-
-        variable = variables[0]
 
-        if variable == 'I':
-            if not u.Unit(axes[variable].unit).is_equivalent('A'):
-                raise Exception("Unit on I axis must be convertible to A.")
+        original_units = {field['name']: field['unit'] for field in self.figure_description.fields}
 
-        if variable == 'j':
-            if not u.Unit(axes[variable].unit).is_equivalent('A / m2'):
-                raise Exception("Unit on j axis must be convertible to A / m².")
+        return self.rescale(original_units)
 
-        return variable
-
-    def x_unit(self, xunit=None):
+    def rescale(self, new_units={}):
```

Comment:
```suggestion
    def rescale(self, units={}):
```

No strong opinion here, but I think "new" is usually not a very helpful prefix.